Support PNPM and Yarn

Right now `next-on-pages` assumes users are using NPM: https://github.com/cloudflare/next-on-pages/blob/main/src/index.ts#L41

Would be nice if this supported PNPM/Yarn as primary package managers too

tried the suggestion above, still doesn't work

```

18:36:44.593 | * branch            e079cdf91ccd7bc7a633cf9c1d21ec04a4af75a2 -> FETCH_HEAD
18:36:44.593 |  
18:36:44.653 | HEAD is now at e079cdf Test timeout with 1 user
18:36:44.653 |  
18:36:44.799 |  
18:36:44.826 | Success: Finished cloning repository files
18:36:45.540 | Installing dependencies
18:36:45.551 | Python version set to 2.7
18:36:47.416 | Attempting node version 'v14.18.3' from .nvmrc
18:36:48.787 | Downloading and installing node v14.18.3...
18:36:49.204 | Downloading https://nodejs.org/dist/v14.18.3/node-v14.18.3-linux-x64.tar.xz...
18:36:49.631 | Computing checksum with sha256sum
18:36:49.759 | Checksums matched!
18:36:54.241 | Now using node v14.18.3 (npm v6.14.15)
18:36:54.641 | Started restoring cached build plugins
18:36:54.654 | Finished restoring cached build plugins
18:36:55.171 | Attempting ruby version 2.7.1, read from environment
18:36:58.746 | Using ruby version 2.7.1
18:36:59.106 | Using PHP version 5.6
18:36:59.272 | 5.2 is already installed.
18:36:59.300 | Using Swift version 5.2
18:36:59.300 | Started restoring cached node modules
18:36:59.317 | Finished restoring cached node modules
18:36:59.564 | Installing NPM modules using NPM version 6.14.15
18:37:07.095 | npm WARN deprecated @floating-ui/react-dom-interactions@0.10.3: Package renamed to @floating-ui/react
18:37:35.745 |  
18:37:35.746 | > prisma@4.9.0 preinstall /opt/buildhome/repo/node_modules/prisma
18:37:35.746 | > node scripts/preinstall-entry.js
18:37:35.746 |  
18:37:36.274 |  
18:37:36.274 | > prisma@4.9.0 install /opt/buildhome/repo/node_modules/prisma
18:37:36.275 | > node scripts/install-entry.js
18:37:36.275 |  
18:37:36.463 |  
18:37:36.463 | > @prisma/engines@4.9.0 postinstall /opt/buildhome/repo/node_modules/@prisma/engines
18:37:36.463 | > node scripts/postinstall.js
18:37:36.464 |  
18:37:37.901 |  
18:37:37.902 | > @prisma/client@4.9.0 postinstall /opt/buildhome/repo/node_modules/@prisma/client
18:37:37.902 | > node scripts/postinstall.js
18:37:37.902 |  
18:37:39.370 | Prisma schema loaded from prisma/schema.prisma
18:37:41.277 |  
18:37:41.277 | ✔ Generated Prisma Client (4.9.0 \| library) to ./node_modules/@prisma/client in 185ms
18:37:41.277 | You can now start using Prisma Client in your code. Reference: ]8;;https://pris.ly/d/clienthttps://pris.ly/d/client]8;;
18:37:41.277 | ```
18:37:41.277 | import { PrismaClient } from '@prisma/client'
18:37:41.278 | const prisma = new PrismaClient()
18:37:41.278 | ```
18:37:41.374 |  
18:37:41.374 | > micohort-suite@0.1.0 postinstall /opt/buildhome/repo
18:37:41.374 | > prisma generate
18:37:41.374 |  
18:37:42.819 | Prisma schema loaded from prisma/schema.prisma
18:37:44.474 |  
18:37:44.475 | ✔ Generated Prisma Client (4.9.0 \| library) to ./node_modules/@prisma/client in 372ms
18:37:44.475 | You can now start using Prisma Client in your code. Reference: ]8;;https://pris.ly/d/clienthttps://pris.ly/d/client]8;;
18:37:44.475 | ```
18:37:44.475 | import { PrismaClient } from '@prisma/client'
18:37:44.475 | const prisma = new PrismaClient()
18:37:44.476 | ```
18:37:44.789 | npm notice created a lockfile as package-lock.json. You should commit this file.
18:37:44.796 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm-eabi@13.1.1 (node_modules/next/node_modules/@next/swc-android-arm-eabi):
18:37:44.796 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm-eabi@13.1.1: wanted {"os":"android","arch":"arm"} (current: {"os":"linux","arch":"x64"})
18:37:44.801 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm64@13.1.1 (node_modules/next/node_modules/@next/swc-android-arm64):
18:37:44.801 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm64@13.1.1: wanted {"os":"android","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:37:44.805 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-arm64@13.1.1 (node_modules/next/node_modules/@next/swc-darwin-arm64):
18:37:44.806 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-arm64@13.1.1: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:37:44.811 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-x64@13.1.1 (node_modules/next/node_modules/@next/swc-darwin-x64):
18:37:44.811 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-x64@13.1.1: wanted {"os":"darwin","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:37:44.815 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-freebsd-x64@13.1.1 (node_modules/next/node_modules/@next/swc-freebsd-x64):
18:37:44.815 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-freebsd-x64@13.1.1: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:37:44.820 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm-gnueabihf@13.1.1 (node_modules/next/node_modules/@next/swc-linux-arm-gnueabihf):
18:37:44.820 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm-gnueabihf@13.1.1: wanted {"os":"linux","arch":"arm"} (current: {"os":"linux","arch":"x64"})
18:37:44.824 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-gnu@13.1.1 (node_modules/next/node_modules/@next/swc-linux-arm64-gnu):
18:37:44.824 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-gnu@13.1.1: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:37:44.828 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-musl@13.1.1 (node_modules/next/node_modules/@next/swc-linux-arm64-musl):
18:37:44.829 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-musl@13.1.1: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:37:44.833 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-arm64-msvc@13.1.1 (node_modules/next/node_modules/@next/swc-win32-arm64-msvc):
18:37:44.833 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-arm64-msvc@13.1.1: wanted {"os":"win32","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:37:44.839 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-ia32-msvc@13.1.1 (node_modules/next/node_modules/@next/swc-win32-ia32-msvc):
18:37:44.840 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-ia32-msvc@13.1.1: wanted {"os":"win32","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
18:37:44.844 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-x64-msvc@13.1.1 (node_modules/next/node_modules/@next/swc-win32-x64-msvc):
18:37:44.844 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-x64-msvc@13.1.1: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:37:44.848 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/chokidar/node_modules/fsevents):
18:37:44.849 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:37:44.855 | npm WARN @emotion/babel-plugin@11.10.5 requires a peer of @babel/core@^7.0.0 but none is installed. You must install peer dependencies yourself.
18:37:44.862 | npm WARN @babel/plugin-syntax-jsx@7.18.6 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
18:37:44.863 |  
18:37:44.866 | added 563 packages from 395 contributors and audited 577 packages in 44.35s
18:37:45.030 |  
18:37:45.030 | 141 packages are looking for funding
18:37:45.030 | run `npm fund` for details
18:37:45.031 |  
18:37:45.031 | found 0 vulnerabilities
18:37:45.031 |  
18:37:45.054 | NPM modules installed
18:37:45.467 | Installing Hugo 0.54.0
18:37:46.157 | Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
18:37:46.160 | Started restoring cached go cache
18:37:46.181 | Finished restoring cached go cache
18:37:46.327 | go version go1.14.4 linux/amd64
18:37:46.342 | go version go1.14.4 linux/amd64
18:37:46.345 | Installing missing commands
18:37:46.346 | Verify run directory
18:37:46.346 | Executing user command: npm i -g pnpm && pnpx @cloudflare/next-on-pages --experimental-minify
18:37:48.186 | /opt/buildhome/.nvm/versions/node/v14.18.3/bin/pnpm -> /opt/buildhome/.nvm/versions/node/v14.18.3/lib/node_modules/pnpm/bin/pnpm.cjs
18:37:48.186 | /opt/buildhome/.nvm/versions/node/v14.18.3/bin/pnpx -> /opt/buildhome/.nvm/versions/node/v14.18.3/lib/node_modules/pnpm/bin/pnpx.cjs
18:37:48.190 | + pnpm@7.25.1
18:37:48.190 | added 1 package in 1.161s
18:37:49.140 | .../share/pnpm/store/v3/tmp/dlx-1878     \| Progress: resolved 1, reused 0, downloaded 0, added 0
18:37:49.707 | .../share/pnpm/store/v3/tmp/dlx-1878     \|  +20 ++
18:37:50.018 | .../esbuild@0.15.18/node_modules/esbuild postinstall$ node install.js
18:37:50.145 | .../share/pnpm/store/v3/tmp/dlx-1878     \| Progress: resolved 42, reused 0, downloaded 20, added 20, done
18:37:50.165 | .../esbuild@0.15.18/node_modules/esbuild postinstall: Done
18:37:50.481 | WARN  Issues with peer dependencies found
18:37:50.482 | .
18:37:50.482 | └─┬ @cloudflare/next-on-pages 0.2.0
18:37:50.482 | └── ✕ missing peer vercel@^28.0.2
18:37:50.482 | Peer dependencies that should be installed:
18:37:50.482 | vercel@^28.0.2
18:37:50.482 |  
18:37:50.745 | ⚡️ @cloudflare/next-to-pages CLI
18:37:50.750 | ⚡️
18:37:50.750 | ⚡️ Installing 'vercel' CLI...
18:37:50.750 | ⚡️
18:38:07.456 | ▲
18:38:07.456 | ▲ > vercel@28.12.8 preinstall /opt/buildhome/repo/node_modules/vercel
18:38:07.457 | ▲ > node ./scripts/preinstall.js
18:38:07.457 | ▲
18:38:07.457 | ▲
18:38:07.641 | ▲
18:38:07.641 | ▲ > esbuild@0.14.47 postinstall /opt/buildhome/repo/node_modules/esbuild
18:38:07.641 | ▲ > node install.js
18:38:07.642 | ▲
18:38:07.642 | ▲
18:38:08.352 | ▲ npm WARN
18:38:08.353 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-32@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-32):
18:38:08.353 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-32@0.14.47: wanted {"os":"linux","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
18:38:08.353 | ▲
18:38:08.357 | ▲ npm WARN
18:38:08.357 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-darwin-64):
18:38:08.358 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-64@0.14.47: wanted {"os":"darwin","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.358 | ▲
18:38:08.361 | ▲ npm WARN
18:38:08.362 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-arm64@0.14.47 (node_modules/esbuild/node_modules/esbuild-android-arm64):
18:38:08.362 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-arm64@0.14.47: wanted {"os":"android","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.362 | ▲
18:38:08.366 | ▲ npm
18:38:08.366 | ▲  WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-arm64@0.14.47 (node_modules/esbuild/node_modules/esbuild-freebsd-arm64):
18:38:08.366 | ▲ npm WARN
18:38:08.366 | ▲ notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-arm64@0.14.47: wanted {"os":"freebsd","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.367 | ▲
18:38:08.370 | ▲ npm WARN
18:38:08.371 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm64@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-arm64):
18:38:08.371 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm64@0.14.47: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.371 | ▲
18:38:08.376 | ▲ npm WARN
18:38:08.377 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-arm):
18:38:08.377 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm@0.14.47: wanted {"os":"linux","arch":"arm"} (current: {"os":"linux","arch":"x64"})
18:38:08.377 | ▲
18:38:08.381 | ▲ npm WARN
18:38:08.381 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-arm64@0.14.47 (node_modules/esbuild/node_modules/esbuild-darwin-arm64):
18:38:08.381 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-arm64@0.14.47: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.382 | ▲
18:38:08.386 | ▲ npm WARN
18:38:08.386 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-freebsd-64):
18:38:08.386 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-64@0.14.47: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.386 | ▲
18:38:08.390 | ▲ npm WARN
18:38:08.390 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-android-64):
18:38:08.390 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-64@0.14.47: wanted {"os":"android","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.391 | ▲
18:38:08.394 | ▲ npm WARN
18:38:08.394 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-openbsd-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-openbsd-64):
18:38:08.394 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-openbsd-64@0.14.47: wanted {"os":"openbsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.394 | ▲
18:38:08.398 | ▲ npm WARN
18:38:08.398 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-mips64le@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-mips64le):
18:38:08.399 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-mips64le@0.14.47: wanted {"os":"linux","arch":"mips64el"} (current: {"os":"linux","arch":"x64"})
18:38:08.399 | ▲
18:38:08.402 | ▲ npm WARN
18:38:08.402 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-arm64@0.14.47 (node_modules/esbuild/node_modules/esbuild-windows-arm64):
18:38:08.402 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-arm64@0.14.47: wanted {"os":"win32","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.402 | ▲
18:38:08.406 | ▲ npm
18:38:08.406 | ▲ WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-riscv64@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-riscv64):
18:38:08.406 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-riscv64@0.14.47: wanted {"os":"linux","arch":"riscv64"} (current: {"os":"linux","arch":"x64"})
18:38:08.407 | ▲
18:38:08.410 | ▲ npm WARN
18:38:08.410 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-netbsd-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-netbsd-64):
18:38:08.411 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-netbsd-64@0.14.47: wanted {"os":"netbsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.411 | ▲
18:38:08.415 | ▲ npm WARN
18:38:08.415 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-32@0.14.47 (node_modules/esbuild/node_modules/esbuild-windows-32):
18:38:08.415 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-32@0.14.47: wanted {"os":"win32","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
18:38:08.415 | ▲
18:38:08.419 | ▲ npm
18:38:08.419 | ▲ WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-sunos-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-sunos-64):
18:38:08.420 | ▲ npm WARN notsup
18:38:08.420 | ▲  SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-sunos-64@0.14.47: wanted {"os":"sunos","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.420 | ▲
18:38:08.424 | ▲ npm WARN
18:38:08.424 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-64@0.14.47 (node_modules/esbuild/node_modules/esbuild-windows-64):
18:38:08.424 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-64@0.14.47: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.425 | ▲
18:38:08.428 | ▲ npm WARN
18:38:08.428 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-s390x@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-s390x):
18:38:08.429 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-s390x@0.14.47: wanted {"os":"linux","arch":"s390x"} (current: {"os":"linux","arch":"x64"})
18:38:08.429 | ▲
18:38:08.432 | ▲ npm WARN
18:38:08.432 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-ppc64le@0.14.47 (node_modules/esbuild/node_modules/esbuild-linux-ppc64le):
18:38:08.433 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-ppc64le@0.14.47: wanted {"os":"linux","arch":"ppc64"} (current: {"os":"linux","arch":"x64"})
18:38:08.433 | ▲
18:38:08.439 | ▲ npm WARN
18:38:08.439 | ▲  @babel/plugin-syntax-jsx@7.18.6 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
18:38:08.440 | ▲
18:38:08.445 | ▲ npm WARN
18:38:08.445 | ▲  @emotion/babel-plugin@11.10.5 requires a peer of @babel/core@^7.0.0 but none is installed. You must install peer dependencies yourself.
18:38:08.446 | ▲
18:38:08.452 | ▲ npm WARN
18:38:08.452 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm-eabi@13.1.1 (node_modules/@next/swc-android-arm-eabi):
18:38:08.453 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm-eabi@13.1.1: wanted {"os":"android","arch":"arm"} (current: {"os":"linux","arch":"x64"})
18:38:08.453 | ▲
18:38:08.459 | ▲ npm WARN
18:38:08.459 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm64@13.1.1 (node_modules/@next/swc-android-arm64):
18:38:08.459 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm64@13.1.1: wanted {"os":"android","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.459 | ▲
18:38:08.465 | ▲ npm WARN
18:38:08.466 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-arm64@13.1.1 (node_modules/@next/swc-darwin-arm64):
18:38:08.466 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-arm64@13.1.1: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.466 | ▲
18:38:08.472 | ▲ npm WARN
18:38:08.473 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-x64@13.1.1 (node_modules/@next/swc-darwin-x64):
18:38:08.473 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-x64@13.1.1: wanted {"os":"darwin","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.473 | ▲
18:38:08.479 | ▲ npm WARN
18:38:08.479 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm-gnueabihf@13.1.1 (node_modules/@next/swc-linux-arm-gnueabihf):
18:38:08.479 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm-gnueabihf@13.1.1: wanted {"os":"linux","arch":"arm"} (current: {"os":"linux","arch":"x64"})
18:38:08.480 | ▲
18:38:08.487 | ▲ npm WARN
18:38:08.487 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-musl@13.1.1 (node_modules/@next/swc-linux-arm64-musl):
18:38:08.487 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-musl@13.1.1: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.488 | ▲
18:38:08.496 | ▲ npm WARN
18:38:08.496 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-gnu@13.1.1 (node_modules/@next/swc-linux-arm64-gnu):
18:38:08.496 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-gnu@13.1.1: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.496 | ▲
18:38:08.503 | ▲ npm WARN
18:38:08.503 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-freebsd-x64@13.1.1 (node_modules/@next/swc-freebsd-x64):
18:38:08.503 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-freebsd-x64@13.1.1: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.503 | ▲
18:38:08.510 | ▲ npm WARN
18:38:08.510 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-arm64-msvc@13.1.1 (node_modules/@next/swc-win32-arm64-msvc):
18:38:08.510 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-arm64-msvc@13.1.1: wanted {"os":"win32","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
18:38:08.510 | ▲
18:38:08.517 | ▲ npm WARN
18:38:08.518 | ▲  optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-ia32-msvc@13.1.1 (node_modules/@next/swc-win32-ia32-msvc):
18:38:08.518 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-ia32-msvc@13.1.1: wanted {"os":"win32","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
18:38:08.518 | ▲
18:38:08.524 | ▲ npm WARN
18:38:08.524 | ▲ optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-x64-msvc@13.1.1 (node_modules/@next/swc-win32-x64-msvc):
18:38:08.525 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-x64-msvc@13.1.1: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"x64"})
18:38:08.525 | ▲
18:38:08.534 | ▲ npm WARN optional
18:38:08.534 | ▲  SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
18:38:08.534 | ▲ npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:38:08.535 | ▲
18:38:08.535 | ▲
18:38:08.535 | ▲
18:38:08.536 | ▲ + vercel@28.12.8
18:38:08.537 | ▲ added 81 packages from 71 contributors and audited 677 packages in 17.107s
18:38:08.537 | ▲
18:38:08.743 | ▲
18:38:08.743 | ▲ 147 packages are looking for funding
18:38:08.743 | ▲   run `npm fund` for details
18:38:08.743 | ▲
18:38:08.743 | ▲
18:38:08.744 | ▲ found 0 vulnerabilities
18:38:08.744 | ▲
18:38:08.744 | ▲
18:38:08.763 | ⚡️
18:38:08.764 | ⚡️
18:38:08.764 | ⚡️ Completed 'npx vercel build'.
18:38:08.764 | ⚡️
18:38:08.764 | ⚡️
18:38:08.764 | ⚡️ Building project with 'npx vercel build'...
18:38:08.765 | ⚡️
18:38:09.322 | ▲ Vercel CLI 28.12.8
18:38:09.322 | ▲
18:38:09.679 | ▲ Detected `pnpm-lock.yaml` generated by pnpm 7...
18:38:09.679 | ▲
18:38:09.698 | ▲ Installing dependencies...
18:38:09.699 | ▲
18:38:10.416 | ▲ Lockfile is up to date, resolution step is skipped
18:38:10.416 | ▲
18:38:10.419 | ▲  ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with package.json
18:38:10.419 | ▲
18:38:10.419 | ▲ Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
18:38:10.419 | ▲
18:38:10.442 | ▲ Error: Command "pnpm install" exited with 1
18:38:10.442 | ▲
18:38:10.508 | (node:1915) UnhandledPromiseRejectionWarning: undefined
18:38:10.508 | (Use `node --trace-warnings ...` to show where the warning was created)
18:38:10.508 | (node:1915) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
18:38:10.508 | (node:1915) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
18:38:10.595 | Finished
18:38:10.596 | Note: No functions dir at /functions found. Skipping.
18:38:10.596 | Validating asset output directory
18:38:10.596 | Error: Output directory ".vercel/output/static" not found.
18:38:11.832 | Failed: build output directory not found

</pre>
```

--- a/.changeset/perfect-pens-breathe.md
+++ b/.changeset/perfect-pens-breathe.md
@@ -1,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+remove redundant vercel install

--- a/src/index.ts
+++ b/src/index.ts
@@ -25,6 +25,9 @@ type LooseNode = Node & {
 };
 
 const prepVercel = async () => {
+  console.log("⚡️");
+  console.log("⚡️ Preparing project for 'npx vercel build'...");
+  console.log("⚡️");
   try {
     await stat(".vercel/project.json");
   } catch {
@@ -35,38 +38,8 @@ const prepVercel = async () => {
     );
   }
   console.log("⚡️");
-  console.log("⚡️ Installing 'vercel' CLI...");
   console.log("⚡️");
-
-  const vercelBuild = spawn("npm", ["install", "-D", "vercel"]);
-
-  vercelBuild.stdout.on("data", (data) => {
-    const lines: string[] = data.toString().split("\n");
-    lines.map((line) => {
-      console.log(`▲ ${line}`);
-    });
-  });
-
-  vercelBuild.stderr.on("data", (data) => {
-    const lines: string[] = data.toString().split("\n");
-    lines.map((line) => {
-      console.log(`▲ ${line}`);
-    });
-  });
-
-  await new Promise((resolve, reject) => {
-    vercelBuild.on("close", (code) => {
-      if (code === 0) {
-        resolve(null);
-      } else {
-        reject();
-      }
-    });
-  });
-
-  console.log("⚡️");
-  console.log("⚡️");
-  console.log("⚡️ Completed 'npx vercel build'.");
+  console.log("⚡️ Project ready for 'npx vercel build'...");
   console.log("⚡️");
 };
 

